### PR TITLE
Added validation rules to podSpec

### PR DESCRIPTION
**Pull request by jshaofuturewei**

### CODE REVIEW

File: pkg/api/testing/backward_compatibility_test.go [hunk 1]
```diff
@@ -48,7 +49,8 @@ func TestCompatibility_v1_PodSecurityContext(t *testing.T) {
 		"containers":[{
 			"name":"a",
 			"image":"my-container-image"
-		}]
+		}],
+        "resourceType": "container"
```
Comment (kimeunju108) [inline pkg/api/testing/backward_compatibility_test.go:53]: In /globalscheduer/test/yaml/sample_20_pods.yaml, resourceType is "vm". Do we need to change "vm" to "container"?
- "resourceType": "container" ==> this file
- "resourceType": "vm" ==> pod yaml file

Comment (jshaofuturewei) [reply pkg/api/testing/backward_compatibility_test.go:53]: "resourceType": "container"  is for non openstack podin the existing test cases
"resourceType": "vm" is for openstack pod in global resource scheduler.

**Review by kimeunju108 — APPROVED** [review]
LGTM

File: pkg/apis/core/validation/validation.go [hunk 2]
```diff
@@ -3182,6 +3185,16 @@ func ValidatePodSpec(spec *core.PodSpec, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, validateInitContainers(true, spec.InitContainers, spec.Containers, vols, fldPath.Child("initContainers"))...)
 		allErrs = append(allErrs, validateWorkloadInfo(true, spec.WorkloadInfo, fldPath.Child("workloadInfo"))...)
 		allErrs = append(allErrs, validateVirtualMachine(spec.VirtualMachine, fldPath.Child("virtualMachine"))...)
+		if spec.ResourceType == "" {
```
Comment (cathyhongzhang) [inline pkg/apis/core/validation/validation.go:3188]: ResourceType can not be "". If it is "", a validation error msg should be output

File: pkg/apis/core/validation/validation.go [hunk 3]
```diff
@@ -3182,6 +3185,16 @@ func ValidatePodSpec(spec *core.PodSpec, fldPath *field.Path) field.ErrorList {
...
+			if spec.VirtualMachine.Flavors != nil && len(spec.VirtualMachine.Flavors) == 0 {
```
Comment (cathyhongzhang) [inline pkg/apis/core/validation/validation.go:3189]: Th logic here is nto quite correct

File: test/integration/etcd/multi_tenancy_data.go [hunk 4]
```diff
@@ -64,15 +64,15 @@ func GetEtcdStorageDataForNamespaceWithMultiTenancy(tenant, namespace string) ma
 			ExpectedEtcdPath: "/registry/configmaps/" + tenant + "/" + namespace + "/cm1",
 		},
 		gvr("", "v1", "services"): {
-			Stub:             `{"metadata": {"name": "service1"}, "spec": {"externalName": "service1name", "ports": [{"port": 10000, "targetPort": 11000}], "selector": {"test": "data"}}}`,
+			Stub:             `{"metadata": {"name": "service1"}, "spec": {"externalName": "service1name", "ports": [{"port": 10000, "tarogetPort": 11000}], "selector": {"test": "data"}}}`,
```
Comment (cathyhongzhang) [inline test/integration/etcd/multi_tenancy_data.go:67]: typo

File: pkg/apis/core/validation/validation.go (hunk 2 above)
Comment (jshaofuturewei) [reply pkg/apis/core/validation/validation.go]: Updated

File: pkg/apis/core/validation/validation.go (hunk 3 above)
Comment (jshaofuturewei) [reply pkg/apis/core/validation/validation.go]: Updated

File: test/integration/etcd/multi_tenancy_data.go (hunk 4 above)
Comment (jshaofuturewei) [reply test/integration/etcd/multi_tenancy_data.go]: Updated

**Review by cathyhongzhang — APPROVED** [review]

**Pull request merged**